src/core/core.scale.js

```diff
@@ -339,7 +340,8 @@ module.exports = Element.extend({
 			var cosRotation, sinRotation;
 
 			// Allow 3 pixels x2 padding either side for label readability
-			var tickWidth = me.getPixelForTick(1) - me.getPixelForTick(0) - 6;
+			var numLabels = labels.length - (tickOpts.offset ? 0 : 1);
+			var tickWidth = (me.width - (margins.left + margins.right)) / numLabels;
```

Comment:
I think both solutions doesn't work as expected when ticks are not at the same distance:

```
labels: [
     '2019-01-02',
     '2019-01-03',
     '2019-02-05',
]

xAxes: [{
   type: time,
   ticks: {
      autoSkip: true,
      source: 'labels'
   }
}]
```

But your proposal may be more accurate

Reply:
I agree, this does not fix the case where ticks are unevenly spaced. It is an more complicated (and uncommon) issue, and should be addressed separately.

Reply:
That comment about 3 pixels is not valid anymore, I should remove it - or it should be accounted in tick generation. Also should look into a regression test. But lets validate the solution first.

Reply:
Totally agree, it should be fixed separately.

`numLabels` will be 0 if `labels.length = 1` and `tickOpts.offset = true`, in which case the next line divides by 0.